testsuite/tests/uart.rs

```diff
@@ -153,14 +223,17 @@ mod tests {
         assert!(matches!(c, Err(Error::Framing)));
 
         // provoke an error (this does not seem to be absolutely deterministic
-        // and we've seen multiple error variants in the wild)
+        // and we've seen multiple error variants in the wild, including
+        // receiving the wrong but valid character)
         unwrap!(nb::block!(tx_fast.write(b'a')));
         let c = nb::block!(rx_slow.read());
         defmt::info!("{}", c);
-        assert!(matches!(c, Err(Error::Framing) | Err(Error::Noise)));
+        assert!(matches!(c, Ok(240) | Err(Error::Framing) | Err(Error::Noise)));
```

Comment:
Hm. Not as reliable as I thought :( Maybe this should be split up and use `defmt::warn!("{}")` in case of `Ok(_) so signal a "flaky" test. The test would still succeed, but it would be obvious while executing, that this is not really the expected output. 

```suggestion
        match c {
            Ok(c) if c != b'a' => {
                defmt::warn!("Unexpected wrong but valid character {}", c);
            },
            Err(Error::Framing) | Err(Error::Noise) => assert!(true),
            _ => assert!(false),
        }
```

Reply:
In my opinion this test is "flaky by nature". As we are sending from fast to slow, the bit transitions sent occur somewhere in between a received bit and I bet we are challenging the USARTs noise detection (see RM _Figure 328., Data sampling when oversampling by 16_). I bet we are on the edge between receiving the wrong character and triggering a noise error. And it's up to some black magic whether the one or the other is the outcome.

Nevertheless, I reformatted the match. Are you okay with having the match as an expression inside `assert!`?

Reply:
> Nevertheless, I reformatted the match. Are you okay with having the match as an expression inside assert! and considering anything but b'a' successful without a warning?

Yes, but I'd like to keep a warning